Projects/Server/Maps/MapSelection.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Server.Maps;
+
+public class MapSelection
+{
+    private MapSelectionFlags _selectedMaps;
+    private static readonly Array _mapSelectionValues = Enum.GetValues(typeof(MapSelectionFlags));
+
+    public static Array MapSelectionValues => _mapSelectionValues;
+
+    public MapSelectionFlags Flags => _selectedMaps;
+
+    public MapSelection()
+    {
+    }
+
+    public MapSelection(MapSelectionFlags flags)
+    {
+        _selectedMaps = flags;
+    }
+
+    public MapSelection(List<string> mapList)
+    {
+        foreach (MapSelectionFlags value in _mapSelectionValues)
+        {
+            if (mapList.Contains(value.ToString()))
+            {
+                Enable(value);
+            }
+        }
+    }
+
+    public bool Includes(MapSelectionFlags mapFlags)
+    {
+        if ((_selectedMaps & mapFlags) == mapFlags)
+        {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
```suggestion
    public bool Includes(MapSelectionFlags mapFlags) => (_selectedMaps & mapFlags) == mapFlags;
```

Reply:
ah yes, still getting used to things like that haha